### PR TITLE
Add PremultiplyAlpha to ResizeOptions

### LINKED ISSUE #1498: Resize: enable opting-out for alpha premultiplication
**Edit:** this turned into a feature request for a `ResizeOptions.PremultiplyAlpha` property defaulting to `true`.

Adding up-for-grabs, since it's easy, here are steps for a potential community PR:
https://github.com/SixLabors/ImageSharp/issues/1498#issuecomment-758564510

----

### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When using Resize(int width, int height) I am getting very odd black bands in the output image. The results seems quite wrong. I am getting a different results when using GIMP's Bicubic Scale Image function.

### Steps to Reproduce
I have attached sample code that results in this issue. I have included the input image in the sample, as well as the output I get on my PC for reference.

### System Configuration

- ImageSharp version: 1.0.2
- Other ImageSharp packages and versions: N/A
- Environment (Operating system, version and so on): I am using Win 10 Pro 2004, with Ryzen 2600, and AMD Radeon R9 380.
- .NET Framework version: .NET 5.0
- Additional information: N/A

Input:
image

My ImageSharp output as seen in GIMP:
image

GIMP's bicubic scale:
image

[ResizeBugProj.zip](https://github.com/SixLabors/ImageSharp/files/5792985/ResizeBugProj.zip)

### PR DESCRIPTION
**Pull request by ptasev**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Added PremultiplyAlpha to the ResizeOptions. This option was always on by default, but now the user has the ability to turn it off through the options.

Fixes #1498

### CODE REVIEW

File: src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs
Comment (ptasev) [inline src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs:180]: Applying compand to the None modifier. Not sure if correct.

File: tests/ImageSharp.Tests/Processing/Processors/Transforms/ResizeTests.cs
Comment (ptasev) [inline tests/ImageSharp.Tests/Processing/Processors/Transforms/ResizeTests.cs:238]: Should there be a ResizeExtension added for quickly specifying PremultiplyAlpha?

Also, here are the reference outputs:
[ResizeTests_PremultiplyAlpha_RefOutput.zip](https://github.com/SixLabors/ImageSharp/files/5806355/ResizeTests_PremultiplyAlpha_RefOutput.zip)

File: src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs
Comment (antonfirsov) [reply src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs:180]: It's OK, although not super happy about how it looks now. Would replace the `ApplyCompanding` helper with something like:

```C#
public static PixelConversionModifiers GetModifiers(bool compand, bool premultiplyAlpha = true);
```

**Review by antonfirsov — APPROVED** [review]
LGTM with a minor suggestion.

Opened SixLabors/Imagesharp.Tests.Images#14 with your images. As soon as it gets merged, we need to push an update of the submodule reference for `tests/Images/External`.

I leave the approval & merge of the submodule PR tor @JimBobSquarePants since we need his final blessing anyways.

Thanks for taking the time!

File: tests/ImageSharp.Tests/Processing/Processors/Transforms/ResizeTests.cs
Comment (antonfirsov) [reply tests/ImageSharp.Tests/Processing/Processors/Transforms/ResizeTests.cs:238]: Since this will be a rarely used option, I prefer not to add new extension methods. There are too many permutations, `ResizeOptions` is here so we can keep them under control.

File: src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs
Comment (ptasev) [reply src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs]: I split it out into a separate private function. I hope that's what you meant.

**Comment by JimBobSquarePants** [comment]
Reference image merged.

```bash
git submodule foreach git pull origin master
```
Will update all the modules.

File: src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs
Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs]: I'm ok with the private function for now. This is the only place that uses it.

**Comment by JimBobSquarePants** [comment]
Thanks for this @ptasev ! I'll get it merged for you now. It'll be available to you in our MyGet feed once built.

**Pull request merged**
